### PR TITLE
Add support for redis `PSUBSCRIBE` command

### PR DESCRIPTION
**Pull request by pavelgnom**
https://redis.io/commands/psubscribe

Added support to subscribe on a channel using pattern.

### CODE REVIEW

File: lib/async/redis/context/psubscribe.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          @connection.write_request ['PSUBSCRIBE', *channels]
+          @connection.flush
+
+          response = nil
+
+          channels.length.times do |i|
+            response = @connection.read_response
+          end
+
+          return response
```
Comment (olleolleolle) [inline lib/async/redis/context/psubscribe.rb]:
```suggestion
          response
```

Can we leave off the `return` statement here?

Comment (pavelgnom) [reply lib/async/redis/context/psubscribe.rb]: In this case I'd suggest to drop it everywhere for consistency.

File: lib/async/redis/context/psubscribe.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        private
+
+        def psubscribe(channels)
+          @connection.write_request ['PSUBSCRIBE', *channels]
+          @connection.flush
+
+          response = nil
+
+          channels.length.times do |i|
```
Comment (olleolleolle) [inline lib/async/redis/context/psubscribe.rb]:
```suggestion
          channels.length.times do
```

Would this do the same thing?

**Comment by ioquatix** [comment]
Sorry, I missed this one. I will review.

**Comment by ioquatix** [comment]
Since this is almost identical to subscribe, do you think it would make sense to make it a subclass? Just wondering if we need to introduce a new surface area for something which is almost the same.

**Pull request closed without merging**